faunadb/client.go

```diff
@@ -235,6 +261,102 @@ func (client *FaunaClient) NewWithObserver(observer ObserverCallback) *FaunaClie
 	return client.newClient(client.basicAuth, observer)
 }
 
+// Stream creates a stream subscription to the result of the given expression.
+//
+// The subscription returned by this method does not issue any requests until
+// the subscription's Start method is called. Make sure to
+// subscribe to the events of interest, otherwise the received events are simply
+// ignored.
+func (client *FaunaClient) Stream(query Expr, config ...StreamConfig) StreamSubscription {
+	return newSubscription(client, query, config...)
+}
+
+func (client *FaunaClient) startStream(subscription *StreamSubscription) (err error) {
+	var payload []byte
+	var response faunaResponse
+
+	startTime := time.Now()
+	payload, err = client.prepareRequestBody(subscription.query)
+	if err != nil {
+		return
+	}
+	body := ioutil.NopCloser(bytes.NewReader(payload))
+
+	var endpoint strings.Builder
+	endpoint.WriteString(client.streamEndpoint)
+
+	if len(subscription.config.Fields) > 0 {
+		endpoint.WriteString("?fields=")
+		count := len(subscription.config.Fields)
+		for i := 0; i < count; i++ {
+			endpoint.WriteString(string(subscription.config.Fields[i]))
+			if i <= count-1 {
+				endpoint.WriteString(",")
+			}
+		}
+	}
+
+	response, err = client.performRequest(body, endpoint.String(), true, nil)
+	if err != nil {
+		return
+	}
+
+	httpResponse := response.response
+	_ = client.storeLastTxnTime(httpResponse.Header)
+	if err = checkForResponseErrors(httpResponse); err != nil {
+		httpResponse.Body.Close()
+		response.cncl()
+		return
+	}
+
+	go func() {
+		<-subscription.closed
+		httpResponse.Body.Close()
+		response.cncl()
+	}()
+
+	go func() {
+		defer func() {
+			httpResponse.Body.Close()
+			response.cncl()
+		}()
+
+		for {
+			var obj Obj
+
+			if val, err := client.parseResponse(httpResponse, subscription.query, true, startTime); err != nil {
+				if err == io.EOF || err.Error() == "http2: response body closed" {
+					break
+				}
+				subscription.events <- ErrorEvent{
+					txn: startTime.Unix(),
```

Comment:
We can't introduce synthetic transaction times. That can lead to consistency issues if users rely on this marker for their domain logic. In this cases, it's best to not report back the a transaction time (you this line and let go assign the vars default value 0).

This change has to be replicated to all `ErrorEvent` initializations. As a guideline, pick the transaction time from the server. If not possible, remove the `txn` and let it be zero.

Reply:
Let's remove timestamp from manually generated events.
Done.